platform/search/bundle.json

```diff
@@ -0,0 +1,34 @@
+{
+    "name": "Search",
+    "description": "Allows the user to search through the file tree.",
+    "extensions": {
+        "constants": [
+            {
+                "key": "GENERIC_SEARCH_ROOTS",
```

Comment:
Since the default value of this is ROOT, I'd say that this _is_ opt-out; you'll be indexing the whole tree unless this constant is overridden elsewhere.

In the real use cases we have, we'll be using a search mechanism associated with a persistence store almost all the time, and generic search in the special case of telemetry dictionaries. It's more natural in that situation to say "do index this part of the tree" than "don't index everything but this part of the tree"